pagopa/io-app
# feat: [IOBP-349] Idpay discount barcode generation result

## Short description
DEPENDS ON https://github.com/pagopa/io-app/pull/5146
<img src="https://github.com/pagopa/io-app/assets/60693085/2974a886-5de3-425e-b849-a11850fe7771" width=200/>

included barcode generation result screen, with all the required additions to make it work as intended.
Some features are still missing, and will be discussed in a first part in the jira issue.

## List of changes proposed in this pull request
- added required I18n keys
- added loading, success and failure mappings and components
- updated navigator and selectors to better comply with the specifications
- added optional fetching for qrcode data, which will be fetched only if expired or null

## How to test
with the `dev-server` running navigate to a discount initiative, then tap on the "authorize a payment" CTA, then "generate barcode". you should check that:
- the screen is correctly rendered
- a failure correctly renders the failure screen, with back navigation leading to the initiative's details
- there is a loading screen that follows design
- the barcode timer keeps track of time past even out of the app
- there are no unneeded renders in the barcode screen, and the `setTimeout` is correctly cleared at the end of the timer or when exiting the component
- when generating multiple barcodes for multiple initiatives opening a non-expired, already generated barcode does not make a new backend request
-  passing time is correctly handled with possible timezone changes
-  when the timer of a barcode runs out, the screen correctly displays a message about the barcode being expired
- overall, check that the screen correctly follows the given specs in the reference jira task.
